cli/flox/src/config/mod.rs

```diff
@@ -497,6 +534,125 @@ mod tests {
         );
     }
 
+    #[test]
+    fn set_by_system_config() {
+        let user_config_dir = tempfile::tempdir().unwrap();
+        let system_config_dir = tempfile::tempdir().unwrap();
+
+        let floxhub_url = "https://testhub.flox.dev";
+        let disable_metrics = true;
+        let search_limit = NonZero::new(42).unwrap();
+
+        fs::write(
+            system_config_dir.path().join(FLOX_CONFIG_FILE),
+            formatdoc! {"
+            floxhub_url = '{floxhub_url}'
+            disable_metrics = {disable_metrics}
+            search_limit = {search_limit}
+        "},
+        )
+        .unwrap();
+
+        // todo: fix mocks to avoid pulling in user config
```

Comment:
I don't completely understand that comment

Reply:
the xdg crate (of the current version) is reading `HOME` and `XDG_*` variables.
It doesn't matter too much with the current tests which take an explicit path now, but testing `locate_*_config_dir` is currently impure and would require mocking different xdg env variables, which id rather do based on the `BaseDirectories` directly with #3313